packages/openapi-v3/src/decorators/parameter.decorator.ts

```diff
@@ -218,8 +218,45 @@ export namespace param {
         ...spec,
       });
     },
+
+    /**
+     * Define a parameter accepting a url encoded string of a JSON object,
+     *
+     * parameter is created with explode: false
+     * 
+     * @param name - Parameter name
+     * @param schema - Optional OpenAPI Schema describing the JSON object.
+     */
+    jsonObject: function(
```

Comment:
The solution is cool! 
Implementation details...I think we don't need a new decorator to support the multi contents, @bajtos 's comment is sth we could try https://github.com/strongloop/loopback-next/pull/4141/files#r347767523.

Or provide the `content` object as the 3rd parameter `spec?: Partial<ParameterObject>`

@deepakrkris You may want to check is it a valid spec to have `schema` and `content` both defined, like:
```
 // new addition - json encoding
content: {
    "application/json": {
         schema
     }
},

// the rest
// @jannyHou: Is it valid to have both `schema` and `content` defined?
 schema,
```

Reply:
And I feel the new signature of `param.query.object()` would be very similar to `@requestBody`, which reminds me to have a chat with @bajtos on https://github.com/strongloop/loopback-next/issues/3493, I will schedule a meeting asap and update our conclusion here. Anyway the spike is good enough to me :) We can determine the signature in the follow-up stories.

Reply:
@jannyHou  I have modified existing decorator with backward compatibility and also fix API Explorer for complex objects in query params.